packages/Kokkos/test/tstKokkosView.cpp

```diff
@@ -0,0 +1,266 @@
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \file   tstKokkosView.cpp
+ * \author Stuart Slattery
+ * \brief  Kokkos view unit tests.
+ */
+//---------------------------------------------------------------------------//
+
+#include "DTK_DBC.hpp"
+
+#include <Kokkos_Core.hpp>
+
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_ScalarTraits.hpp>
+
+#include <type_traits>
+
+//---------------------------------------------------------------------------//
+// TEST FUNCTORS
+//---------------------------------------------------------------------------//
+// Fill a view with an index.
+template<class View>
+class FillFunctor
+{
+  public:
+
+    FillFunctor( View data )
+        : _data( data )
+    { /* ... */ }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const size_t i) const 
+    { _data(i) = static_cast<typename View::traits::value_type>(i); }
+
+  private:
+
+    View _data;
+};
+
+//---------------------------------------------------------------------------//
+// Assign one 2D view to another.
+template<class View1, class View2>
+class AssignFunctor
+{
+  public:
+
+    AssignFunctor( View1 view_1, View2 view_2 )
+        : _view_1(view_1)
+        , _view_2(view_2)
+    {
+        static_assert( 
+            std::is_same<typename View1::traits::value_type,
+                         typename View2::traits::value_type>::value,
+            "View data types must be the same" );
+        static_assert( 
+            std::is_same<typename View1::traits::device_type,
+                         typename View2::traits::device_type>::value,
+            "View device types must be the same" );
+        static_assert( static_cast<unsigned>(View1::Rank) == 2,
+                       "View ranks must be 2" );
+        static_assert( static_cast<unsigned>(View2::Rank) == 2,
+                       "View ranks must be 2" );
+        DTK_REQUIRE( view_1.extent(0) == view_2.extent(0) );
```

Comment:
Good point. I did not realize they are going to deprecate `dimension_0()` and friends in favor of `extent`. I think they did that to align more with upcoming C++ standard.